src/main/java/uk/gov/hmcts/reform/finrem/documentgenerator/service/DocumentConversionService.java

```diff
@@ -0,0 +1,100 @@
+package uk.gov.hmcts.reform.finrem.documentgenerator.service;
+
+
+
+
+import com.google.common.io.Files;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.hmcts.reform.finrem.documentgenerator.error.DocumentConversionException;
+import uk.gov.hmcts.reform.finrem.documentgenerator.model.Document;
+
+import java.io.File;
+import java.io.IOException;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class DocumentConversionService {
+
+    private static final String PDF = "pdf";
+
+    @Value("${service.pdf-service.uri}/rs/convert")
+    private String documentConversionUrl;
+
+    @Value("${service.pdf-service.accessKey}")
+    private String docmosisAccessKey;
+
+    private final RestTemplate restTemplate;
+
+    private final EvidenceManagementService evidenceManagementService;
+
+
+    public byte[] convertDocumentToPdf(Document sourceDocument) {
+        if (sourceDocument.getFileName().toLowerCase().endsWith(PDF)) {
```

Comment:
I'm not convinced checking document extension is a valid way of checking the file type. We already have Apache Tika in DocGenService that can be leveraged for that purpose: https://github.com/hmcts/finrem-document-generator-client/blob/master/src/main/java/uk/gov/hmcts/reform/finrem/documentgenerator/config/DocumentGeneratorApplicationConfiguration.java#L12